lib/lightning_web/controllers/collections_controller.ex

```diff
@@ -66,55 +77,177 @@ defmodule LightningWeb.CollectionsController do
+    with {:ok, collection} <- Collections.get_collection(col_name),
          :ok <- authorize(conn, collection) do
-      conn =
-        conn
-        |> put_resp_content_type("application/json")
-        |> send_chunked(200)
-
-      collection_items = collection_stream(collection)
-
-      # Take a stream of items (they enter the stream in batches of @query_all_limit)
-      # we then chunk them into groups of 20 and send them to the client
-      # since a chunk for each line seems a bit too many
-      # we can play/adjust this number to see what works best
-      wrap_array(collection_items, &Jason.encode!/1)
-      |> Stream.chunk_every(20)
-      |> Enum.reduce_while(conn, fn chunk, conn ->
-        case Plug.Conn.chunk(conn, chunk) do
-          {:ok, conn} ->
-            {:cont, conn}
-
-          {:error, :closed} ->
-            {:halt, conn}
-        end
-      end)
+      case Collections.put(collection, key, value) do
+        :ok ->
+          json(conn, %{key: key, error: nil})
+
+        {:error, :not_foud} ->
+          conn
+          |> put_status(:not_found)
+          |> json(%{key: key, error: "Item Not Found"})
+      end
     end
   end
 
-  defp wrap_array(stream, encode_func) do
-    Stream.concat([
-      ["["],
-      stream
-      |> Stream.map(fn item -> encode_func.(item) end)
-      |> Stream.intersperse(","),
-      ["]"]
-    ])
-  end
-
-  defp collection_stream(collection) do
-    Stream.unfold(nil, fn cursor ->
-      Repo.transaction(fn ->
-        Collections.stream_all(collection, cursor)
-        |> Enum.to_list()
-      end)
-      |> case do
-        {:ok, []} -> nil
-        {:ok, items} -> {items, items |> List.last() |> Map.get(:updated_at)}
+  def put_all(conn, %{"name" => col_name, "items" => items}) do
+    with {:ok, collection} <- Collections.get_collection(col_name),
+         :ok <- authorize(conn, collection) do
+      case Collections.put_all(collection, items) do
+        {:ok, count} ->
+          json(conn, %{upserts: count, error: nil})
+
+        :error ->
+          conn
+          |> put_status(:internal_server_error)
+          |> json(%{upserts: 0, error: "Database Error"})
       end
+    end
+  end
+
+  def get(conn, %{"name" => col_name, "key" => key}) do
+    with {:ok, collection} <- Collections.get_collection(col_name),
+         :ok <- authorize(conn, collection) do
+      case Collections.get(collection, key) do
+        %Item{} = item ->
+          json(conn, item)
+
+        nil ->
+          conn
+          |> put_status(:not_found)
+          |> json(%{key: key, error: "Item Not Found"})
+      end
+    end
+  end
+
+  def delete(conn, %{"name" => col_name, "key" => key}) do
+    with {:ok, collection} <- Collections.get_collection(col_name),
+         :ok <- authorize(conn, collection) do
+      case Collections.delete(collection, key) do
+        :ok ->
+          json(conn, %{key: key, error: nil})
```

Comment:
As discussed on the spec, can we have a single delete look just like a matched delete:

```
{ deleted: 101, keys: [...]}
```

I don't know if we actually need a 404 here? I think `{ deleted: 0, keys: [] }` is clear enough.

Thoughts @josephjclark?